core/src/test/scala/org/apache/spark/sql/delta/rowtracking/DefaultRowCommitVersionSuite.scala

```diff
@@ -0,0 +1,158 @@
+          commitVersionForFiles += a.path -> commitVersion
+        case r: RemoveFile if commitVersionForFiles.contains(r.path) =>
+          assert(r.defaultRowCommitVersion.contains(commitVersionForFiles(r.path)))
+        case _ =>
+          // Do nothing
+      }
+    }
+    commitVersionForFiles.toMap
+  }
+
+  test("defaultRowCommitVersion is not set when feature is disabled") {
+    withRowTrackingEnabled(enabled = false) {
+      withTempDir { tempDir =>
+        spark.range(start = 0, end = 100, step = 1, numPartitions = 1)
+          .write.format("delta").mode("overwrite").save(tempDir.getAbsolutePath)
+        spark.range(start = 100, end = 200, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+
+        val deltaLog = DeltaLog.forTable(spark, tempDir)
+        deltaLog.update().allFiles.collect().foreach { f =>
+          assert(f.defaultRowCommitVersion.isEmpty)
+        }
+      }
+    }
+  }
+
+  test("checkpoint preserves defaultRowCommitVersion") {
+    withRowTrackingEnabled(enabled = true) {
+      withTempDir { tempDir =>
+        spark.range(start = 0, end = 100, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+        spark.range(start = 100, end = 200, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+        spark.range(start = 200, end = 300, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+
+        val deltaLog = DeltaLog.forTable(spark, tempDir)
+        val commitVersionForFiles = expectedCommitVersionsForAllFiles(deltaLog)
+
+        deltaLog.update().allFiles.collect().foreach { f =>
+          assert(f.defaultRowCommitVersion.contains(commitVersionForFiles(f.path)))
+        }
+
+        deltaLog.checkpoint(deltaLog.update())
+
+        deltaLog.update().allFiles.collect().foreach { f =>
+          assert(f.defaultRowCommitVersion.contains(commitVersionForFiles(f.path)))
+        }
+      }
+    }
+  }
+
+  test("data skipping reads defaultRowCommitVersion") {
+    withRowTrackingEnabled(enabled = true) {
+      withTempDir { tempDir =>
+        spark.range(start = 0, end = 100, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+        spark.range(start = 100, end = 200, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+        spark.range(start = 200, end = 300, step = 1, numPartitions = 1)
+          .write.format("delta").mode("append").save(tempDir.getAbsolutePath)
+
+        val deltaLog = DeltaLog.forTable(spark, tempDir)
+        val commitVersionForFiles = expectedCommitVersionsForAllFiles(deltaLog)
+
+        val filters: Seq[Expression] = Seq(col("id = 150").expr)
+        val scan: DeltaScan = deltaLog.update().filesForScan(filters)
```

Comment:
Is there some reason these `val` (and only these) need explicit types?

Reply:
There's no reason. I think I temporarily added them when I was fighting with IntellIJ not being able to infer the right type. I will remove it.